core/src/main/java/org/mule/runtime/core/api/processor/strategy/ProcessingStrategy.java

```diff
@@ -47,17 +49,13 @@
     return publisher -> from(publisher).transform(pipelineFunction);
   }
 
-
   /**
    * Enrich {@link Processor} function by adding pre/post operators to implement processing strategy behaviour.
    *
-   * @param processor processor instance.
-   * @param processorFunction processor function
    * @return enriched processor function
    */
-  default Function<Publisher<Event>, Publisher<Event>> onProcessor(Processor processor,
-                                                                   Function<Publisher<Event>, Publisher<Event>> processorFunction) {
-    return publisher -> from(publisher).transform(processorFunction);
+  default Function<ReactiveProcessor, ReactiveProcessor> onProcessor() {
```

Comment:
Yes, let's do that.  Just revert PS changes and use `Function<Publisher<Event>, Publisher<Event>>` instead of `ReativeProcessor` in `AbstractMessageProcessorChain`